x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTaskIndexService.java

```diff
@@ -225,7 +227,7 @@ public void deleteResponse(AsyncExecutionId asyncExecutionId,
                                ActionListener<DeleteResponse> listener) {
         try {
             DeleteRequest request = new DeleteRequest(index).id(asyncExecutionId.getDocId());
-            client.delete(request, listener);
+            createIndexIfNecessary(ActionListener.wrap(v -> client.delete(request, listener), listener::onFailure));
```

Comment:
Do we need to create an index in this case or this is for consistency?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/64606/commits/a273d920cd9f91e67507abe2c1c321c3e9eb62ad to explain the reasoning.

Reply:
:+1: